openssl/openssl.go

```diff
@@ -126,28 +123,33 @@ func loadLibrary(version string) (unsafe.Pointer, error) {
 		}
 		return handle, nil
 	}
-	for _, v := range knownVersions {
+	fallbackIdx := -1
+	for i, v := range knownVersions {
 		handle := dlopen(v)
+		if handle == nil {
+			continue
+		}
+		if C.go_openssl_fips_enabled(handle) == 1 {
+			// Found a FIPS enabled version, use it.
+			return handle, nil
+		}
+		C.dlclose(handle)
+		if fallbackIdx == -1 {
+			// Remember the first version that exists but is not FIPS enabled
+			// in case we don't find any FIPS enabled version.
+			fallbackIdx = i
```

Comment:
This would only optimize an edge case and may be more complicated than it's worth, but seems like this could potentially avoid an extra `dlclose`/`dlopen` by storing the fallback `handle` and only closing it when we're sure we don't need it.

Reply:
Not complicated all all, will do!